### ISSUE TITLE
dev_c_refactor: gdb client crashes on exit

### ISSUE THREAD

**Issue by snickerbockers** [washingtondc-emu/washingtondc#5]
In theory this is gdb's fault for not handling whatever non-clean exit the GdbStub is doing, but I still should fix it before I merge this branch to master.  886722ea05f55d0f236e562cdef21249bc371e0c is the commit that introduced the problem:
```
jay@sbckrs_desktop ~/programs/washingtondc $ git bisect bad
886722ea05f55d0f236e562cdef21249bc371e0c is the first bad commit
commit 886722ea05f55d0f236e562cdef21249bc371e0c
Author: snickerbockers <chimerasaurusrex@gmail.com>
Date:   Tue Apr 25 05:57:00 2017 -0700

    port SerialServer and GdbStub from boost.asio to libevent

:100644 100644 5cbd084ba32103aa54bb2331b40366c18cd03de5 b4012cc0cc7159b96264c22be28b94ab34b9b277 M	CMakeLists.txt
:040000 040000 96f36f27ddc699755df7f06dafb6e85749eb3db3 44f98f7a7ef9586b40000ed9487bf0b8fa1f22cd M	src
```

**Comment by snickerbockers** [comment]
I'm still not sure what boost.asio is doing differently from libevent, but I am now certain that this is a problem with gdb because the documentation says that stubs are allowed to immediately close the connection upon receiving a K packet without sending a response or even acknowledging the packet.  I have also noticed that this problem is reproducible on gdb 7.11 but not gdb 7.10.1.

One other datapoint is that if I have wash itself running in gdb and I set a break-point for gdb_cleanup, then the remote gdb session disconnects normally when I step through the call to bufferevent_free.
